micronota/bfillings/diamond.py

```diff
@@ -230,19 +231,30 @@ def _annotate_fp(self, fp, aligner='blastp', evalue=0.001, cpus=1,
             elif outfmt == 'sam':
                 res = res.append(
                     self._filter_id_cov(self.parse_sam(out_fp)))
-            found.extend(res.index)
+
             # save to a tmp file the seqs that do not hit current database
             new_fp = join(self.tmp_dir, '%s.fa' % out_prefix)
+            num_left = 0
+
+            found = found | set(res.index)
             with open(new_fp, 'w') as f:
                 for seq in read(fp, format='fasta'):
                     if seq.metadata['id'] not in found:
                         seq.write(f, format='fasta')
-                        seqs.append(seq)
+                    num_left += 1
+            logger.info('Number of diamond hits: %d' % len(res.index))
+            logger.info('Number of sequence left unmatched: %d' % num_left)
```

Comment:
not crucial. can always open the output to find the total number of CDS.
let's just keep one.

On Sun, Mar 27, 2016 at 7:16 PM, Jamie Morton <notifications@github.com>
wrote:

> In micronota/bfillings/diamond.py
> <https://github.com/biocore/micronota/pull/71#discussion_r57539937>:
>
> >              with open(new_fp, 'w') as f:
> >                  for seq in read(fp, format='fasta'):
> >                      if seq.metadata['id'] not in found:
> >                          seq.write(f, format='fasta')
> > -                        seqs.append(seq)
> > +                    num_left += 1
> > +            logger.info('Number of diamond hits: %d' % len(res.index))
> > +            logger.info('Number of sequence left unmatched: %d' % num_left)
>
> Possibly? But are we keeping track of how many hits there are to begin
> with?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/micronota/pull/71/files/ea04d76e442d51faa50c21eabc6d5c5d56ec0459#r57539937>
>